medusa/notifiers/join.py

```diff
@@ -0,0 +1,119 @@
+        return self._sendjoin(
+            join_api=join_api,
+            join_device=join_device,
+            event='Test',
+            message='Testing Join settings from Medusa',
+            force=True
+        )
+
+    def notify_snatch(self, ep_name, is_proper):
+        """Send Join notification when nzb snatched if selected in config."""
+        if app.JOIN_NOTIFY_ONSNATCH:
+            self._sendjoin(
+                join_api=None,
+                event=common.notifyStrings[(common.NOTIFY_SNATCH, common.NOTIFY_SNATCH_PROPER)[is_proper]] + ' : ' + ep_name,
+                message=ep_name
+            )
+
+    def notify_download(self, ep_name):
+        """Send Join notification when nzb download completed if selected in config."""
+        if app.JOIN_NOTIFY_ONDOWNLOAD:
+            self._sendjoin(
+                join_api=None,
+                event=common.notifyStrings[common.NOTIFY_DOWNLOAD] + ' : ' + ep_name,
+                message=ep_name
+            )
+
+    def notify_subtitle_download(self, ep_name, lang):
+        """Send Join notification when subtitles downloaded if selected in config."""
+        if app.JOIN_NOTIFY_ONSUBTITLEDOWNLOAD:
+            self._sendjoin(
+                join_api=None,
+                event=common.notifyStrings[common.NOTIFY_SUBTITLE_DOWNLOAD] + ' : ' + ep_name + ' : ' + lang,
+                message=ep_name + ': ' + lang
+            )
+
+    def notify_git_update(self, new_version='??'):
+        """Send Join notification when new version available from git."""
+        self._sendjoin(
+            join_api=None,
+            event=common.notifyStrings[common.NOTIFY_GIT_UPDATE],
+            message=common.notifyStrings[common.NOTIFY_GIT_UPDATE_TEXT] + new_version,
+        )
+
+    def notify_login(self, ipaddress=''):
+        """Send Join notification when login detected."""
+        self._sendjoin(
+            join_api=None,
+            event=common.notifyStrings[common.NOTIFY_LOGIN],
+            message=common.notifyStrings[common.NOTIFY_LOGIN_TEXT].format(ipaddress)
+        )
+
+    def _sendjoin(self, join_api=None, join_device=None, event=None, message=None, force=False):
+        """Compose and send Join notification."""
+        push_result = {'success': False, 'error': ''}
+
+        if not (app.USE_JOIN or force):
+            return False
+
+        join_api = join_api or app.JOIN_API
+        join_device = join_device or app.JOIN_DEVICE
+        icon_url = 'https://cdn.pymedusa.com/images/ico/favicon-310.png'
+
+        log.debug('Join title: {0!r}', event)
+        log.debug('Join message: {0!r}', message)
+        log.debug('Join api: {0!r}', join_api)
+        log.debug('Join devices: {0!r}', join_device)
+
+        post_data = {'title': event, 'text': message, 'deviceId': join_device, 'apikey': join_api, 'icon': icon_url}
+        urllib.quote_plus = urllib.quote
+        params = urllib.urlencode(post_data)
```

Comment:
No need to urlencode the dict, if your using requests.get with params=.

Reply:
I think I had trouble doing it that way. I can do another test to see, it's been a while since I wrote that but I do remember something weird with the way join wanted the string encoded.

Reply:
Yep, you were correct. Removed unneeded url encoding. Not sure why I didn't get it working the first time I tried. Thanks for pointing that out.